WT-12669 Add compressed and uncompressed values for backup stats.

sueloverso opened the pull request:
I added these stats to distinguish between collections and indexes. They are in addition to `backup_blocks`. I considered removing that and just having the two. Let me know if you have an opinion on that.

keitharnoldsmith reviewed (approved):
LGTM.

Two questions/thoughts. Neither requires a change, unless you think we should followup on them (and they could be follow-up tickets.

1. I don't think the statistics tell us the incremental backup block size. If a user wants to convert between the blocks backed up and the bytes backed up, they need to go find the block size. So we should report whatever we think users are most likely to want to reason about. But I'm not sure which that is (or even if it makes any real difference).
2. When a user creates a new table between backups, I think the incremental backup cursor tells them to just copy the entire file. Is that (should that) be captured by these statistics?

sueloverso commented:
@keitharnoldsmith I added the granularity stat, and I added per-table block counts per @ajmorton . I am not going to include whole-file copying information in this ticket because I'm not sure how that would help the user at the moment.

keitharnoldsmith commented on src/cursor/cur_backup_incr.c:
```diff
@@ -202,6 +205,10 @@ __curbackup_incr_next(WT_CURSOR *cursor)
+                if (F_ISSET(cb, WT_CURBACKUP_COMPRESSED))
+                    WT_STAT_CONN_INCR(session, backup_blocks_compressed);
+                else
+                    WT_STAT_CONN_INCR(session, backup_blocks_uncompressed);
```
Should these two be `WT_STAT_CONN_DATA_INCR()`?

sueloverso replied:
Refresh. I caught that and fixed.

keitharnoldsmith replied:
Thanks Sue! And there's a test to go with it. Nice. 

My previous LGTM still stands :-)

keitharnoldsmith reviewed (approved):
LGTM -- possible bug, but the fix is straight-forward, so I don't need to re-review.

ajmorton commented on dist/stat_data.py:
```diff
@@ -954,6 +964,12 @@ def __init__(self, name, desc, flags=''):
+    ##########################################
+    # Backup statistics
+    ##########################################
+    BackupStat('backup_blocks_compressed', 'total modified incremental blocks in tables with block compression'),
```
nit: These stats are per-table but the descriptions say tables plural

ajmorton commented on test/suite/wtbackup.py:
```diff
@@ -319,10 +326,36 @@ def take_incr_backup_block(self, bkup_c, newfile, backup_incr_dir, consolidate):
+                uncomp = self.backup_get_stat(stat.dsrc.backup_blocks_uncompressed, newfile)
+                comp_diff = comp - last_comp
+                uncomp_diff = uncomp - last_uncomp
+                if blocks_diff == 0:
+                    self.assertEqual(0, comp_diff + uncomp_diff)
+                else:
+                    self.assertNotEqual(0, comp_diff + uncomp_diff)
```
Could this become
```suggestion
                self.assertEqual(blocks_diff, comp_diff + uncomp_diff)
```
?
It looks like we perform the same check a few lines below, but only when `did_work` is true

sueloverso replied:
I originally had that and although all the backup tests passed, the compact tests that do backup were failing. I'll investigate more what compact is doing that causes the failure.

sueloverso replied:
Fixed

ajmorton reviewed (approved):
LGTM. Thanks!
One nit and one suggestion, neither of which need re-review

sueloverso commented:
The failing test is WT-12430, a known issue unrelated to my changes. Merging.

ajmorton commented:
@sueloverso after the migration from `evergreen merge` to `merge queue` all tests need to pass before PRs will be added to the merge queue. I've restarted the failing test on the assumption it's flaky and will pass on a second run.

sueloverso commented:
@etienneptl it appears that my backup statistics changes has made hitting WT-12430 a lot more common. I don't think my backup stats changes are causing the problem, but my repros are all on this branch and apparently not limited to MacOS.

sueloverso commented:
@etienneptl the last 2-3 pushes and PR testing hit the `compact11` issue. However, the last one passed. So it isn't 100% hitting just often.

etienneptl commented:
@sueloverso, @Sean04 and I are looking into it and will give you an update asap.

etienneptl commented:
@sueloverso, WT-12430 is being merged. It was tested with your changes and the test stopped failing. Feel free to merge `develop` once WT-12430 is completed.
